src/middlewared/middlewared/plugins/virt/attachments.py

```diff
@@ -212,6 +214,31 @@ async def get_ports(self):
         return ports
 
 
+class IncusServicePortDelegate(ServicePortDelegate):
+
+    name = 'incus'
+    namespace = 'incus'
+    title = 'Incus Service'
+
+    async def get_ports_internal(self):
+        ports = []
+        try:
+            # Get incusbr0 network details from incus API
+            network_info = await self.middleware.call('virt.global.get_network', INCUS_BRIDGE)
```

Comment:
It seems like `INCUS_BRIDGE` should not be used here, it only should be used if `virt.global.config` has `bridge` unset. @Qubad786 can you confirm please ?

Otherwise what ever bridge we have configured there should be used instead.
Also i think there is no need to do this if virt is not setup, so we can easily get that information first

Reply:
> Otherwise what ever bridge we have configured there should be used instead.
> Also i think there is no need to do this if virt is not setup, so we can easily get that information first

I'm not sure how I can get this details. But makes sense, feel free to push changes if you think it will be faster/easier than to give me pointers!